cscs-checks/tools/jupyter/jupyter_tests.py

```diff
@@ -9,32 +9,36 @@
 
 @rfm.simple_test
 class JupyterHubSubmitTest(rfm.RunOnlyRegressionTest):
-    def __init__(self):
-        self.valid_systems = ['daint:jupyter_gpu', 'daint:jupyter_mc',
-                              'dom:jupyter_gpu', 'dom:jupyter_mc']
-        self.valid_prog_environs = ['*']
-        self.sourcesdir = None
-        self.executable = 'hostname'
-        self.time_limit = '1m'
-        self.max_pending_time = '7m'
+    valid_systems = ['daint:jupyter_gpu', 'daint:jupyter_mc',
+                     'dom:jupyter_gpu', 'dom:jupyter_mc']
+    valid_prog_environs = ['*']
+    sourcesdir = None
+    executable = 'hostname'
+    time_limit = '1m'
+    max_pending_time = '7m'
+    tags = {'production', 'post-maintenance', 'health'}
+    maintainers = ['RS', 'TR']
+
+    @rfm.run_before('sanity')
+    def set_sanity_check(self):
         self.sanity_patterns = sn.assert_found(r'nid\d+', self.stdout)
-        self.tags = {'production', 'post-maintenance', 'health'}
-        self.maintainers = ['RS', 'TR']
 
 
 @rfm.simple_test
 class JupyterHubServerCheck(rfm.RunOnlyRegressionTest):
-    def __init__(self):
-        self.descr = 'Check JupyterHub server status and version'
-        self.valid_systems = ['daint:jupyter_gpu', 'daint:jupyter_mc',
-                              'dom:jupyter_gpu', 'dom:jupyter_mc']
-        self.valid_prog_environs = ['*']
-        self.sourcesdir = None
-        self.executable = 'curl https://jupyter.cscs.ch/hub/api/'
-        self.time_limit = '30s'
-        self.num_tasks = 1
-        self.num_tasks_per_node = 1
+    descr = 'Check JupyterHub server status and version'
+    valid_systems = ['daint:jupyter_gpu', 'daint:jupyter_mc',
+                     'dom:jupyter_gpu', 'dom:jupyter_mc']
+    valid_prog_environs = ['*']
+    sourcesdir = None
+    executable = 'curl https://jupyter.cscs.ch/hub/api/'
```

Comment:
Do we want to execute this from the Jupyter compute nodes or from the login nodes?

Reply:
In principle I would say from compute nodes, but we could assume that if the server is reachable from a login node then it is reachable also from the nodes in the jupyter partition.